src/inmanta/server/validate_filter.py

```diff
@@ -183,3 +185,64 @@ def process_filters(self, filter: Dict[str, List[str]]) -> Dict[str, QueryFilter
             query["action"] = (QueryType.CONTAINS, validated_filter.action)
 
         return query
+
+
+def parse_single(v: object, property_name: str) -> object:
+    """
+    Transform list values to their single element value.
+    """
+    if isinstance(v, list):
+        return more_itertools.one(
+            v,
+            too_short=ValueError(f"Empty '{property_name}' filter provided"),
+            too_long=ValueError(f"Multiple values provided for '{property_name}' filter: {v}"),
+        )
+
+    return v
+
+
+class CompileReportFilterModel(FilterModel):
+    requested: Optional[DateRangeConstraint]
+    started: Optional[bool]
+    completed: Optional[bool]
+    success: Optional[bool]
+
+    @validator("started", pre=True)
+    @classmethod
+    def _started_single(cls, v: object) -> object:
+        return parse_single(v, "started")
+
+    @validator("completed", pre=True)
+    @classmethod
+    def _completed_single(cls, v: object) -> object:
+        return parse_single(v, "completed")
+
+    @validator("success", pre=True)
+    @classmethod
+    def _success_single(cls, v: object) -> object:
+        return parse_single(v, "success")
+
+    @validator("requested", pre=True)
+    @classmethod
+    def parse_requested(cls, v: object) -> List[Tuple[RangeOperator, datetime.datetime]]:
+        return parse_range_operator(v)
+
+
+class CompileReportFilterValidator(FilterValidator[CompileReportFilterModel]):
```

Comment:
We could make this a lot more declarative by having a dict, that indicates for every field what type of filter it is

```python

class CompileReportFilterValidator(FilterValidator[CompileReportFilterModel]):
     allowed_filter = {
                  "requested": DateRangeFilter,
                  "success": BooleanEqualityFilter,
                  "started": BooleanNoneIsFalse,
                  "completed":BooleanNoneIsFalse,
     }
     
``` 

and validate purely on that